GEOSldas_App/util/postproc/ObsFcstAna_stats/postproc_ObsFcstAna.py

```diff
@@ -0,0 +1,392 @@
+                        
+                        for var in var_list:
+                            masked_data[var] = OFA[var][is_valid]
+              
+                        for var in var_list:
+                            data_tile[var][tile_idx, ispec] = masked_data[var]         
+
+                data_all.append(data_tile)
+
+            # cross-mask over all experiments 
+            is_cross_valid = ~np.isnan(data_all[0]['obs_obs'])
+            for data in data_all[1:]:
+                mask = ~np.isnan(data['obs_obs'])
+                is_cross_valid = np.logical_and(is_cross_valid,mask)
+
+            # reconstruct the output variable dictionary based on input options;
+            # obs_obs and obs_obsvar are from exp_list[obs_from], the rest are from exp_list[0]
+            data_tile = {}    
+            for var in var_list:
+                if 'obs_obs' in var:
+                    data_tile[var] = data_all[obs_from][var]
+                else:
+                    data_tile[var] = data_all[0][var]
+
+            is_valid = is_cross_valid
+            
+            N_data[ is_valid] += 1
+            oxf_sum[is_valid] += data_tile['obs_obs' ][is_valid] * data_tile['obs_fcst'][is_valid]
+            oxa_sum[is_valid] += data_tile['obs_obs' ][is_valid] * data_tile['obs_ana' ][is_valid]
+            fxa_sum[is_valid] += data_tile['obs_fcst'][is_valid] * data_tile['obs_ana' ][is_valid]
+
+            for var in var_list:
+                data_sum[ var][is_valid] += data_tile[var][is_valid]
+                data2_sum[var][is_valid] += data_tile[var][is_valid] **2
+            
+            date_time = date_time + timedelta(seconds=da_dt)
+
+        return N_data, data_sum, data2_sum, oxf_sum, oxa_sum, fxa_sum
+
+    # ----------------------------------------------------------------------------------------------------------
+    #
+    # Function to compute monthly sums and save results in nc4 files for all months in [start_time, end_time].
+    #
+    # Skips computation/saving of monthly sums output if file already exists.
+
+    def save_monthly_sums(self):
+        expdir_list    = self.expdir_list
+        expid_list     = self.expid_list
+        exptag_list    = self.exptag_list
+        domain         = self.domain
+        var_list       = self.var_list
+        tc             = self.tilecoord
+        obsparam_list  = self.obsparam_list
+
+        date_time   = self.start_time
+
+        while date_time < self.end_time           # loop through months
+            
+            # make monthly file output directory 
+            mo_path = self.sum_path + '/Y'+ date_time.strftime('%Y') + '/M' + date_time.strftime('%m') + '/'
+            os.makedirs(mo_path, exist_ok=True)
+    
+            fout = self.outid + '.ens_avg.ldas_ObsFcstAna.' + date_time.strftime('%Y%m') +'_sums.nc4'
+
+            fout = mo_path + fout
+            
+            # skip if output file already exists
```

Comment:
I ran out of time to look into this in more detail, but are we ensuring that an existing sums file is consistent with new sums files generated for other months in terms of exp_list (i.e., cross-masking and "obs_from")?

Reply:
Ok, thanks.  This should work if we have no more than 2 or maybe 3 experiments in exp_list, and if the exp names are relatively short.  Otherwise the file names of the sums files might become too long.  Let's see if what we have now works for Andy.  We might still need to add the "config" to each nc4 file.  This should make it straightforward to read "config" from an existing monthly sums file and verify it against the "config" of the current run.